src/main/java/core/basesyntax/CheckElectricCar.java

```diff
@@ -1,8 +1,10 @@
 package core.basesyntax;
 
-/**
- * <p>Имплементируйте необходимый функциональный интерфейс и его метод, который реализует проверку
- * установлен ли на авто EngineType.ELECTRIC. Метод должен принимать Car.</p>
- */
-public class CheckElectricCar {
+import java.util.function.Predicate;
+
+public class CheckElectricCar implements Predicate<Car> {
+    @Override
+    public boolean test(Car car) {
+        return car.getEngine().getEngineType() == EngineType.ELECTRIC;
```

Comment:
maybe its better to use equals with objects? but im not sure))

Reply:
If I understand correctly, here it doesn`t matter because we haven't override equals, so we call default Object equals() which working next way this == obj